### ISSUE TITLE
[1.16 Beta Deployment] Error in logic for parsing file extension

### ISSUE THREAD

**Issue by Maurier** [hydroshare/hydroshare#2947]
https://github.com/hydroshare/hydroshare/blob/develop/theme/static/js/hs_file_browser.js#L253

File extension detection logic needs to be fixed to include "."
The current implementation erroneously enables and disables functions based on file name.

For example, Zip function is enabled for a file called "readmezip" which is not actually a .zip file.
![image](https://user-images.githubusercontent.com/2448568/47247415-aded6d00-d3c0-11e8-8e51-e847c8803301.png)

**Comment by engineSound** [comment]
It is a simple fix as you pointed out. Just change to endsWith(".ZIP")

https://github.com/hydroshare/hydroshare/blob/0e44d6c9b96e8b1736064ca6b13c88220e739263/theme/static/js/hs_file_browser.js#L253

**Comment by Maurier** [comment]
@engineSound yes, this should be included in 1.16 since this bug can cause unexpected behavior for functionality regarding unzipping and setting logical file types. Adding "." to each case should fix it.

**Comment by engineSound** [comment]
I will fix the issue. This code seemed to be like this for a few releases. If we change all the extension files, we test all the file extensions. We need to run this by @sblack-usu 

I found a workaround. I can rename the file abcZIP to abc.zip using the rename 
![image](https://user-images.githubusercontent.com/35352595/47259174-df734080-d473-11e8-883e-7c6f183b39dc.png) 
, the unzip icon was enabled.

**Comment by dtarb** [comment]
I do not feel this should be a 1.16 show stopper.  But it would be good to fix if it is simple.  I was not able to reproduce on www so it does seem to be a regression introduced in beta.

**Comment by engineSound** [comment]
The code change is little but require a large scale testing on all the file formats. Without automation testing in place to support regression testing, I don't recommend rush this into 1.16 to cause regressions.

**Comment by sblack-usu** [comment]
This is a pretty minor frontend javascript change.  I don't see how it requires extensive testing as a user can manually test the scope of this change.  Also, it should be noted this is not only a problem with zip, but any other extension that is reserved for aggregations.  The code just checks the end of the filename instead of getting the extension and checking that.  This looks like it was something I did when we started checking for refts.json, so the old file extension logic didn't work.  I clearly did not think it through to include the "."  I believe we can get this deployed and tested for 1.16.
